packages/edit-navigation/src/components/header/index.js

```diff
@@ -2,40 +2,97 @@
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { Button, Dropdown } from '@wordpress/components';
+import {
+	Spinner,
+	Button,
+	SelectControl,
+	Dropdown,
+} from '@wordpress/components';
 
 /**
  * Internal dependencies
  */
 import ManageLocations from './manage-locations';
+import AddMenuForm from './add-menu-form';
 
-export default function Header( { onBeginAddingMenu } ) {
+export default function Header( {
+	menus,
+	selectedMenuId,
+	onSelectMenu,
+	onCancelAddingMenu,
+} ) {
 	return (
 		<div className="edit-navigation-header">
 			<h1>{ __( 'Navigation' ) }</h1>
 
-			<Button
-				className="edit-navigation-header__add-new-button"
-				isSecondary
-				onClick={ onBeginAddingMenu }
-			>
-				{ __( 'Add new' ) }
-			</Button>
-
-			<Dropdown
-				contentClassName="edit-navigation-header__manage-locations"
-				position="bottom center"
-				renderToggle={ ( { isOpen, onToggle } ) => (
-					<Button
-						isLink
-						aria-expanded={ isOpen }
-						onClick={ onToggle }
-					>
-						{ __( 'Manage locations' ) }
-					</Button>
+			<div className="edit-navigation-header__actions">
+				{ menus ? (
+					<div className="edit-navigation-header__current-menu">
+						<p>{ __( 'Currently editing' ) }</p>
+						<SelectControl
+							className="edit-navigation-toolbar__menu-select"
+							label={ __( 'Currently editing' ) }
+							hideLabelFromVision
+							disabled={ ! menus.length }
+							value={ selectedMenuId ?? 0 }
+							options={
+								menus.length
+									? menus.map( ( menu ) => ( {
+											value: menu.id,
+											label: menu.name + ' (Primary)', //Just a reminder to show the menu location, if set.
+									  } ) )
+									: [
+											{
+												value: 0,
+												label: __(
+													'— Select navigation —'
+												),
+											},
+									  ]
+							}
+							onChange={ onSelectMenu }
+						/>
+					</div>
+				) : (
+					<Spinner />
```

Comment:
It's not ideal that the `SelectControl` "pops in" after load.

Reply:
I don't mind it, but if you've got a better solution I'd love it.

Reply:
I removed the spinner in 417ee5502e. I still don't really like how the width of the dropdown changes during page load, but we can come back to this later.